### ISSUE TITLE
Error message "The file “mimjdbc3.jar” couldn’t be opened."

### ISSUE THREAD

**Issue by majorproblem** [glimpseio/Splice#10]
I tried to create a JDBC connection to the database my company builds, Mimer SQL, but ran into problems.

I created a new connection and selected JDBC from the list of databases and then filled out the subsequent window as follows:

Driver: /opt/mimersql1101-11.0.1A/lib/mimjdbc3.jar
Driver class: com.mimer.jdbc.Driver
Driver URL: jdbc:mimer:tcp://localhost:1360/mimerdb
Username: SYSADM
Password: SYSADM

I then clicked the "Test" button and got the following response: Successfully connected to mimer:tcp://localhost:1360/mimerdb

However, when I then clicked "Save", instead of saving this connection I received a message dialogue stating that "The file “mimjdbc3.jar” couldn’t be opened.".

Since I obviously could open the file when I tested the connection I am rather perplexed about this.

**Comment by marcprux** [comment]
When you enter in the driver jar path, are you selecting it by clicking the folder button next to the  "Driver" parameter, or are you pasting the path into the text field? There is a known problem with entering raw text into that field, so if you select the file with the folder button, it should persist the driver jar path.

If that doesn't help, can you let us know if the /opt/ path is special in some way, for example a  mounted network drive path or a symbolic link? You can also try moving the jar file to your "Downloads" folder and selecting it from there; permissions for accessing files in that directory are a bit more lax for sandboxed apps that other paths.

Lastly, please try all this with Splice 1.0.2, which was just released to the App Store, and so should be available for you to install shortly.

If none of that works, can you send us your .splice project file, either by attaching it to this issue or mailing it to support@glimpse.io?

**Comment by majorproblem** [comment]
I clicked the button and navigated to /usr/local/ where the driver resides, and that resolves to the /opt path. I have not entered anything manually.

The /opt structure is not special in any way:

```
macos-high-sierra--10:opt appleseed$ ls -la
total 0
drwxr-xr-x   3 root  wheel   96 May 31 10:06 .
drwxr-xr-x  31 root  wheel  992 Jul 17 09:13 ..
drwxrwxr-x   9 root  wheel  288 May 31 10:05 mimersql1101-11.0.1A
```
and further down
```
macos-high-sierra--10:opt appleseed$ ls -la mimersql1101-11.0.1A/
total 0
drwxrwxr-x   9 root  wheel   288 May 31 10:05 .
drwxr-xr-x   3 root  wheel    96 May 31 10:06 ..
drwxrwxr-x  55 root  wheel  1760 May 31 10:05 bin
drwxrwxr-x   9 root  wheel   288 May 31 10:05 doc
drwxrwxr-x  21 root  wheel   672 May 31 10:05 examples
drwxrwxr-x   9 root  wheel   288 May 31 10:05 include
drwxrwxr-x  15 root  wheel   480 May 31 10:05 lib
drwxrwxr-x   5 root  wheel   160 May 31 10:05 man
drwxrwxr-x  12 root  wheel   384 May 31 10:05 misc
```
I will try the 1.0.2 version presently… :-)

**Comment by majorproblem** [comment]
The same error persists with 1.0.2, I'm sad to report. 😞

**Comment by majorproblem** [comment]
I have finally managed to get this to work and it seems as if your original hunch about permissions is indeed correct, even it I have not a clue as to what and where.

I ended up downloading our driver from our development site, rather that using the driver that comes with the database distribution. That file was expanded to a small folder with the driver and the documentation inside the Downloads folder.

I then created a JDBC connection using that jar-file instead and, lo and behold, it works!
